add-on/src/lib/state.js

```diff
@@ -40,8 +41,15 @@ export function initState (options, overrides) {
       return false
     }
   }
-  // TODO state.connected ~= state.peerCount > 0
-  // TODO state.nodeActive ~= API is online,eg. state.peerCount > offlinePeerCount
+  // TODO refactor this into a class. It's getting too big and messy.
+  Object.defineProperty(state, 'connected', {
+    // TODO: make quick fetch to confirm it works?
+    get: function () { return this.peerCount > offlinePeerCount + 1 }
+  })
+  Object.defineProperty(state, 'nodeActive', {
```

Comment:
Do you know if things defined via `Object.defineProperty`  will survive `postMessage` serialization present in places where we use `browser.runtime.connect` ?

I vaguely remember that we did not do getters like this in the past because it did not survive serialization (maybe browser fixed it these days?). That is why we read `peerCount` field  directly (without getters) and compare it to `offlinePeerCount` const.

By adding these getters here, we are having two ways of doing online check in the codebase, which over time leads to errors.

My suggestion is to remove these `Object.defineProperty` to keep this PR scope small, and if you want to add them, do that in separate PR that also refactors all places which act on `peerCount`  (if possible, but  I feel it may not be worth the work).